benchmarks/README.md

```diff
@@ -0,0 +1,99 @@
+# PermaplanT Benchmarks
+
+## Requirements
+
+-   nodejs 19.4.0
+-   npm
+
+## Installation and Usage
+
+1. Install dependencies
+
+```shell
+npm install
+```
+
+2. Populate the database with example maps
+
+In order to benchmark the performance of the canvas, we need to populate the database with example maps.
+There are maps of different sizes and complexity, which are stored in a single CSV file called `maps.csv` in the `/benchmarks` folder.
+The maps are stored in a CSV file because they are huge and it is easier to store them in a CSV file than in a database migrations,
+To populate the database with the maps, run the following command:
+
+```shell
+npm run start
+```
+
+Note: This will clean the `maps` table and insert the maps from the CSV file.
+
+3. Start the backend from the backend folder
+
+```shell
+cargo run
+```
+
+4. Start the frontend from the frontend folder
+
+```shell
+npm run dev
+```
+
+5. Run benchmarks
+
+```shell
+npm run benchmark
+```
+
+## Implementation
+
+### Backend
+
+-   `maps` table is simple and contains only the map name and the canvas as a JSON string.
+
+-   Simple backend implementation for map creation and retrieval.
+
+### Frontend
+
+-   The frontend implementation i.e. react state management is also simple and doesn't contain any complex logic except the canvas itself.
+
+-   Available canvas functionality:
+    -   save map
+    -   hide layers
+    -   generate circles on a new layer
+    -   generate rectangles on a new layer
+    -   undo/redo
+    -   zoom in/out
+    -   reset zoom
+    -   drag and drop
+
+### Benchmarking
+
+The benchmarking script runs a performance audit on a web page using `Lighthouse` and `Playwright`.
+The audit measures the performance of a web page by generating a performance score and then saves the results of the audit in `report` folder:
+
+-   `<test_name>-report.json` - the raw report of LightHouse for a single test case (e.g. `100 circles` on the canvass)
+-   `<timestamp>-lighthouse-results.csv` - the results of the audit for all test cases (e.g. `100 circles`, `1000 circles`, `10000 circles` etc.)
+-   `<timestamp>-lighthouse-results-shapes.csv` - the results of the audit for all test cases related to shapes (e.g. `100 circles` vs. `100 stars`).
+
+Metrics:
+
+-   `First Contentful Paint (FCP)` - the time it takes for the browser to render the first bit of content on the page. Measured in milliseconds.
+-   `Interactive` - the time it takes for the page to become fully interactive. Measured in milliseconds.
+-   `Score` - the performance score of the page. Measured in percentage. The metric is calculated based on the weighted average of the other metrics. (PENDING, will probably be removed because it depends on the other metrics, which are not considered in the study)
```

Comment:
Where do I see it? Which values did you get? In the playright report I have e.g.:

```
10000 circles with 10 layer
12msBefore Hooks
23msbrowser.newPage— canvas_performance.spec.js:15
20mspage.goto(http://localhost:5173/demo/7)— canvas_performance.spec.js:16
29mspage.evaluate— ../node_modules/playwright-lighthouse/src/audit.js:40
35msAfter Hooks
```
stdout:

```
Starting Lighthouse audit for 10000 circles with 10 layer


-------- playwright lighthouse audit reports --------


performance record is 62 and desired threshold was 0
Lighthouse report saved to report/10000 circles with 10 layer-report.json
Lighthouse results saved to report/1680358909532-lighthouse-results.csv
```

Reply:
The outputs are well defined just above the lines you commented on:

> The audit measures the performance of a web page by generating a performance score and then saves the results of the audit in `report` folder:

I'm not sure why you've been looking in another folder, maybe just overseen the lines. Let me know if it was still unclear to you where to find the results and what is the structure of the benchmarking. And I will change the README based on your suggestion

https://github.com/ElektraInitiative/PermaplanT/pull/94/files/d113eeec53f081a34bda0c7beea5cc90ff459cc4#diff-576c27794bae284efc278336a16d0649aaf599973a64d4b65dc67bc99bb7f000R72

Reply:
Probably there is just some problem with terminology. What is the difference between "performance score", "performance" (as in the*-lighthouse-results.csv) and "score" (as defined in the commented line)? Anyway, you should first clarify for yourself which score actually is the most useful and better describe what it is.